torchbenchmark/models/gen_torchvision_benchmarks.py

```diff
@@ -7,38 +7,62 @@
     if not os.path.isdir(folder):
         os.makedirs(folder)
     input_shape = (32, 3, 224, 224)
-    example_inputs = f'(torch.randn({input_shape}),)'
+    example_inputs = f'(torch.randn({input_shape}).to(self.device),)'
     eval_inputs = 'example_inputs[0][0].unsqueeze(0)'
     init_program = f"""
 # Generated by gen_torchvision_benchmark.py
 import torch
 import torch.optim as optim
 import torchvision.models as models
+from ...util.model import BenchmarkModel
+from torchbenchmark.tasks import COMPUTER_VISION
 
-class Model:
-    def __init__(self, device="cpu", jit=False):
+#######################################################
+#
+#       DO NOT MODIFY THESE FILES DIRECTLY!!!
+#       USE `gen_torchvision_benchmarks.py`
+#
+#######################################################
+class Model(BenchmarkModel):
+    task = COMPUTER_VISION.CLASSIFICATION
+    def __init__(self, device=None, jit=False):
+        super().__init__()
         self.device = device
         self.jit = jit
-        self.model = models.{class_model}()
+        self.model = models.{class_model}().to(self.device)
+        self.eval_model = models.{class_model}().to(self.device)
+
         if self.jit:
             self.model = torch.jit.script(self.model)
+            self.eval_model = torch.jit.script(self.eval_model)
+            # model needs to in `eval`
+            # in order to be optimized for inference
+            self.eval_model.eval()
+            self.eval_model = torch.jit.optimize_for_inference(self.eval_model)
         self.example_inputs = {example_inputs}
 
     def get_module(self):
         return self.model, self.example_inputs
 
+    # vision models have another model
+    # instance for inference that has
+    # already been optimized for inference
+    def set_eval(self):
```

Comment:
May I ask what is the purpose of this function?

Reply:
`set_eval` is defined `BenchmarkModel` and by default it runs `get_module()[0].eval()` and it's overriden for the models that don't use `get_module()` for `test_eval()`. Here's I'm just overriding it for clarity to do nothing since we have a whole new model instance for inference.